provider/internal/device/ftd/ftdonboarding/resource.go

```diff
@@ -0,0 +1,158 @@
+package ftdonboarding
+
+import (
+	"context"
+	"fmt"
+	cdoClient "github.com/CiscoDevnet/terraform-provider-cdo/go-client"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var _ resource.Resource = &Resource{}
+
+func NewResource() resource.Resource {
+	return &Resource{}
+}
+
+type Resource struct {
+	client *cdoClient.Client
+}
+
+type ResourceModel struct {
+	Id    types.String `tfsdk:"id"`
+	FtdId types.String `tfsdk:"ftd_id"`
+}
+
+func (r *Resource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_ftd_device_onboarding"
+}
+
+func (r *Resource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "TODO_FILL_ME",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "TODO_FILL_ME",
```

Comment:
What does id represent in the context of the FTD onboarding resource?

Reply:
I set `id` as the registration key, because I felt like there should be one FTD onboarding resource per registration key, should it be per CDO FTD instead (i.e. use FTD uid)? Or maybe per FTD specific device (specific device uid)